features/step_definitions/functional_tests/functional_tests_steps.rb

```diff
@@ -0,0 +1,59 @@
+Given(/^the test package "([^"]*)" is installed on the controller node$/) do |test_package|
+  control_node.rpm_q(test_package)
+end
+
+Given(/^the package "([^"]*)" is installed on the controller node$/) do |client_package|
+  control_node.rpm_q(client_package)
+end
+
+Given(/^the proper cirros test image has been created$/) do
+  test_image_name = "cirros-test-image-uec"
+
+  if control_node.openstack.image.list.find {|img| img.name == test_image_name }
+    control_node.openstack.image.delete(test_image_name)
```

Comment:
I would skip that. If there is a image with that name, we can assume it is the correct one. And we want to reuse the image for all tests if possible. Deleting and add new images with the same name is imo just confusing.